File: week01_08_17/PRG_다음큰숫자_홍윤식.java
class Solution {
    public int solution(int n) {
        int answer = n+1;// 정답은 n보다 큰수
        int nCnt=0;//1의 갯수
      
		//n을 2진수로 바꿨을 때 1의 갯수 세기
		for (int i = 0; i < 19; i++) {	
			if ((n & 1 << i) != 0) {
				nCnt++;
			}
		}
        while(true){
            int temp =answer;
            int tempCnt=0;
			for (int i = 0; i < 19; i++) {	
				if ((temp & 1 << i) != 0) {
					tempCnt++;
				}
			}
            
            if(tempCnt==nCnt){
                return answer;
            }
            answer++;
        }
        
    }
}

File: week01_08_17/PRG_다음큰숫자_홍윤식.py
# def solution(n):
#     answer = 0
#     cnt = bin(n).count('1')
#     while True:                                                         
#         n=n+1
#         if bin(n).count('1')==cnt:
#             answer=n
#             break
#     return answer

def solution(n):
    answer=n+1
    oneCnt=0;
    for i in range(19):
        if n&1<<i!=0:
            oneCnt+=1
    while True:
        tempCnt=0
        for i in range(19):
            if answer&1<<i!=0:
                tempCnt+=1
        if tempCnt==oneCnt:
            return answer
        answer+=1